python_modules/dagster/dagster/_core/execution/plan/compute_generator.py

```diff
@@ -143,7 +143,13 @@ def _validate_multi_return(
     context: OpExecutionContext,
     result: Any,
     output_defs: Sequence[OutputDefinition],
-) -> None:
+) -> Any:
+    # special cases for implicit/explicit returned None
+    if result is None:
+        # extrapolate None -> (None, None, ...) for multi assets that do not subset
+        if context.has_assets_def and not context.assets_def.can_subset:
```

Comment:
> Might it also make sense to support this for non-asset with multiple outputs?

for sure, this current code is motivated by whats needed in the stack but from the previous inline discussion my leading contender is

> interpret implicit or explicit return None to mean return (None, None, ...) for any multiple output asset or op. Disallow when there are "optional" outputs to prevent control flow misinterpretation.

Ill update this PR to that approach soon